File: spring-data-requery/src/main/java/org/springframework/data/requery/converters/LocalTimeToLongConverter.java
package org.springframework.data.requery.converters;

import io.requery.Converter;

import javax.annotation.Nullable;
import java.time.LocalTime;

/**
 * org.springframework.data.requery.converters.LocalTimeToLongConverter
 *
 * @author debop
 */
public class LocalTimeToLongConverter implements Converter<LocalTime, Long> {

    @Override
    public Class<LocalTime> getMappedType() {
        return LocalTime.class;
    }

    @Override
    public Class<Long> getPersistedType() {
        return Long.class;
    }

    @Nullable
    @Override
    public Integer getPersistedSize() {
        return Long.SIZE;
    }

    @Override
    public Long convertToPersisted(LocalTime value) {
        return (value != null) ? value.toNanoOfDay() : null;
    }

    @Override
    public LocalTime convertToMapped(Class<? extends LocalTime> type, @Nullable Long value) {
        return (value != null) ? LocalTime.ofNanoOfDay(value) : null;
    }
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/converters/LocalTimeToLongConverter.kt
package org.springframework.data.requery.converters

import io.requery.Converter
import java.time.LocalTime

/**
 * org.springframework.data.requery.converters.LocalTimeToLongConverter
 * @author debop
 * @since 18. 7. 2
 */
class LocalTimeToLongConverter: Converter<LocalTime, Long> {

    override fun getPersistedType(): Class<Long> = Long::class.java
    override fun getMappedType(): Class<LocalTime> = LocalTime::class.java
    override fun getPersistedSize(): Int? = null

    override fun convertToMapped(type: Class<out LocalTime>?, value: Long?): LocalTime? {
        return value?.let { LocalTime.ofNanoOfDay(it) }
    }

    override fun convertToPersisted(value: LocalTime?): Long? {
        return value?.toNanoOfDay()
    }


}